src/main/java/com/github/junit5docker/DockerExtension.java

```diff
@@ -6,90 +6,69 @@
 import org.junit.jupiter.api.extension.BeforeEachCallback;
 import org.junit.jupiter.api.extension.ExtensionContext;
 
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.concurrent.TimeUnit;
-import java.util.function.Supplier;
-import java.util.stream.Stream;
-
-import static java.util.concurrent.CompletableFuture.supplyAsync;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
 
 class DockerExtension implements BeforeAllCallback, AfterAllCallback, BeforeEachCallback, AfterEachCallback {
 
     private final DockerClientAdapter dockerClient;
 
-    private String containerId;
+    private final DockerLogChecker dockerLogChecker;
```

Comment:
Awesome refactoring ! Makes the code way simpler :) 👍

Reply:
thanks! it was motivated by checkstyle/pmd saying the class had too many dependencies :)